bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/BookieShell.java

```diff
@@ -2050,6 +2061,57 @@ protected void readLedgerIndexEntries(long ledgerId) throws IOException {
     }
 
     /**
+     * Read ledger index entries
+     * 
+     * @param ledgerId
+     * @return
+     * @throws IOException if failed to init KeyValueStorageRocksDB
+     */
+    protected int readLedgerIndexEntriesFromDbLedgerStorage(long ledgerId) throws IOException {
+
+        ServerConfiguration conf = new ServerConfiguration(bkConf);
+        LedgerDirsManager ledgerDirsManager = new LedgerDirsManager(bkConf, bkConf.getLedgerDirs());
+
+        String ledgersPath = ledgerDirsManager.getAllLedgerDirs().get(0).toString();
+        String locationsDbPath = FileSystems.getDefault().getPath(ledgersPath, "locations").toFile().toString();
+        KeyValueStorageRocksDB ledgersDb;
+        try {
+            ledgersDb = new KeyValueStorageRocksDB(locationsDbPath, DbConfigType.Small, conf, true);
+        } catch (IOException e) {
+            System.err.printf("ERROR: initializing rocksDb storage %s", e.getMessage());
+            return -1;
+        }
+        try {
+            byte[] array = new byte[16];
+            for (long curEntry = 0; curEntry <= Long.MAX_VALUE; curEntry++) {
+                ArrayUtil.setLong(array, 0, ledgerId);
+                ArrayUtil.setLong(array, 8, curEntry);
+                byte[] value = new byte[8];
+                try {
+                    int code = ledgersDb.get(array, value);
+                    if (code <= 0) {
+                        // no-more entry found
+                        System.out.println("entry " + curEntry + "\t:\tN/A");
+                        break;
+                    }
+                    long offset = ArrayUtil.getLong(value, 0);
+                    long entryLogId = offset >> 32L;
+                    long pos = offset & 0xffffffffL;
+                    System.out.println("entry " + curEntry + "\t:\t(log:" + entryLogId + ", pos: " + pos + ")");
+                } catch (Exception e) {
+                    System.err.println("ERROR: while getting location for entry " + ledgerId + ", " + curEntry + ", "
+                            + e.getMessage());
+                    break;
+                }
+            }
+        } finally {
+            ledgersDb.close();
+        }
```

Comment:
This feels way too much like you are reaching into the internals of EntryLocationIndex.  Even though we do very similar things in other parts of this same file, I really would prefer to at least see the internals of most of this moving to EntryLocationIndex.java, so the code that messes with this DB is all in the same file.

In fact it looks like you could use

https://github.com/yahoo/bookkeeper/blob/yahoo-4.3/bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/storage/ldb/EntryLocationIndex.java#L77-L94

for almost all of this, and no need for accessing ArrayUtil